py-polars/polars/utils/udfs.py

```diff
@@ -3,10 +3,20 @@
 import os
 from typing import Any
 
+from polars._utils.deprecation import issue_deprecation_warning
+
 __all__ = ["_get_shared_lib_location"]
 
 
 def _get_shared_lib_location(main_file: Any) -> str:
+    issue_deprecation_warning(
+        "polars.utils will be made private in a future release. Please use "
+        "`from polars.plugins import register_plugin` instead. "
+        "Note that its interface has changed - check the user guide "
+        "(https://docs.pola.rs/user-guide/expressions/plugins) "
+        "for the currently-recommended way to register a plugin.",
```

Comment:
The `utils` module being made private isn't really relevant to the user. We should tell them to use the new `register_plugin` function, which will auto-detect the plugin location if the library root path is passed.